wasmtime-jni/src/io_github_kawamuray_wasmtime_Module/imp.rs

```diff
@@ -12,6 +12,70 @@ pub(super) struct JniModuleImpl;
 const OBJECT_CLASS: &'static str = "java/lang/Object";
 const LIMIT_TYPE: &str = "io/github/kawamuray/wasmtime/MemoryType$Limit";
 pub const IMPORT_TYPE_CLASS: &'static str = "io/github/kawamuray/wasmtime/ImportType$Type";
+pub const EXPORT_TYPE_CLASS: &'static str = "io/github/kawamuray/wasmtime/ExportType$Type";
+pub const STRING_CLASS: &str = "java/lang/String";
+
+macro_rules! match_extern_type {
```

Comment:
This doesn't need to be a macro, and could be just a function with a signature like: `fn match_extern_type<'a>(env: &'a JNIEnv, ty: ExternType) -> Result<(JObject<'a>, JObject<'a>)>`, as we're going to have single `ExternType` for both imports and exports (even not, the function can take a function as an argument with signature lie `FnOnce(&'a JNIEnv, &str) -> Result<JObject<'a>>` to generate type object).

Reply:
FnOnce is quite a great type! That makes it way better than macros. But you're right, with one type there is no need for either.